src/types/package.rs

```diff
@@ -450,6 +482,107 @@ pub struct PackageDescriptor {
     pub package_type: PackageType,
 }
 
+/// Describes a package descriptor in the system with lockfile path
+#[derive(
+    PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Debug, Serialize, Deserialize, JsonSchema,
```

Comment:
I don't think it's good that types in `phylum-types` derive `JsonSchema`. We have code here in `phylum-types` that is a dependency of both the frontend and the backend, but we don't have separate versioning going on for this crate. The `JsonSchema` attribute indicates that this type is being returned from the backend, but since it's also consumed directly by the frontend, we end up with circular dependency problems when we try to make updates to these types.

I think ideally we would define types that are used to talk to the backend in the backend and autogenerate the client, but we're not doing it. In some places we have code in `phylum-types` that _used_ to be used by both the backend and the frontend but the backend has had to duplicate it to handle compatibility problems.